src/test/java/uk/gov/hmcts/reform/sscs/personalisation/SyaAppealCreatedAndReceivedPersonalisationTest.java

```diff
@@ -598,6 +598,46 @@ public void givenAnAppealWithRepresentativeAndAppellantHasTwoOtherParties_setOth
         assertTrue(result.get(AppConstants.YOUR_DETAILS_LITERAL).contains("Child maintenance number: 123456"));
     }
 
+    @Test
+    public void givenAnAppealWithOtherPartyWithEmptyNameAndEmptyAddress_setOtherPartiesDetailsForTemplate() {
```

Comment:
This unit test doesn't look like a test for the changes made above.

Reply:
How come? There was no test added to cover the 'Not provided' stuff before and it was actually throwing a NPE when we had an empty address. So this basically covers that missing scenario.